binrw/src/binread/mod.rs

```diff
@@ -46,7 +46,7 @@ pub trait BinRead: Sized + 'static {
     /// [`read()`]: Self::read
     /// [`read_args()`]: Self::read_args
     /// [`read_options()`]: Self::read_options
-    type Args: Clone;
+    type Args<'a>;
```

Comment:
Dropping Clone here scares me does this not break everything?

Reply:
# ![All checks have passed](https://user-images.githubusercontent.com/229244/204102521-b78fbc06-aa1f-4a5a-b7f6-ce5c784bda00.png) !!!!!!!!!!!! :-)

This PR moves the constraint down to the functions where the `Clone` trait is actually used. At the moment, this is anything that calls `after_parse`, and anything that creates a collection using the same arguments for each item. Other places don’t require it, so having the constraint on `Args` is needlessly restrictive.

I don’t know why I didn’t think about it before, but actually `after_parse` signature should just be changed so it doesn’t take the arguments any more, and anything that actually does something with an argument in `after_parse` should just take and store that data inside `read_options`. Then, the only thing that ever needs `Args: Clone` are the collection types/helpers, and the generation of `after_parse` is much simpler. I’ll do this in a separate patch.